modules/transactionpool/transactionpool.go

```diff
@@ -9,86 +9,53 @@ import (
-// from using them.
-type TransactionPool struct {
-	// Depedencies of the transaction pool. The state height is needed
-	// separately from the state because the transaction pool may not be
-	// synchronized to the state.
-	consensusSet       modules.ConsensusSet
-	gateway            modules.Gateway
-	consensusSetHeight types.BlockHeight
-
-	// A linked list of transactions, with a map pointing to each. Incoming
-	// transactions are inserted at the tail if they do not conflict with
-	// existing transactions. Transactions pulled from reverted blocks are
-	// inserted at the head because there may be dependencies. Inserting in
-	// this order ensures that dependencies always appear earlier in the linked
-	// list, so a call to TransactionSet() will never dump out-of-order
-	// transactions.
-	transactions    map[crypto.Hash]struct{}
-	transactionList []types.Transaction
-
-	// The unconfirmed set of contracts and outputs. The unconfirmed set
-	// includes the confirmed set, except for elements that have been spent by
-	// the unconfirmed set.
-	siacoinOutputs map[types.SiacoinOutputID]types.SiacoinOutput
-	fileContracts  map[types.FileContractID]types.FileContract
-	siafundOutputs map[types.SiafundOutputID]types.SiafundOutput
-
-	// The reference set contains any objects that are not in the unconfirmed
-	// set, but may still need to be referenced when creating diffs or
-	// reverting unconfirmed transactions (due to conflicts).
-	referenceSiacoinOutputs        map[types.SiacoinOutputID]types.SiacoinOutput
-	referenceFileContracts         map[types.FileContractID]types.FileContract
-	referenceFileContractRevisions map[crypto.Hash]types.FileContract
-	referenceSiafundOutputs        map[types.SiafundOutputID]types.SiafundOutput
-
-	// The entire history of the transaction pool is kept. Each element
-	// represents an atomic change to the transaction pool. When a new
-	// subscriber joins the transaction pool, they can be sent the entire
-	// history and catch up properly, and they can take a long time to catch
-	// up. To prevent deadlocks in the transaction pool, subscribers are
-	// updated in a separate thread which does not guarantee that a subscriber
-	// is always fully synchronized to the transaction pool.
-	consensusChangeIndex    int   // Increments any time a consensus update is made.
-	consensusChanges        []int // The index of the consensus change from the consensus set. -1 means there was no change from the consensus set.
-	unconfirmedTransactions [][]types.Transaction
-	unconfirmedSiacoinDiffs [][]modules.SiacoinOutputDiff
-	subscribers             []chan struct{}
-
-	mu *sync.RWMutex
-}
+type (
+	// ObjectIDs are the IDs of objects such as siacoin outputs and file
+	// contracts, and are used to see if there are conflicts or overlaps within
+	// the transaction pool. A TransactionSetID is the hash of a transaction
+	// set.
+	ObjectID         crypto.Hash
```

Comment:
it is generic though. There's probably a better name but I haven't yet thought of it.